services/api/src/utils/search.js

```diff
@@ -45,15 +47,17 @@ function getSearchQuery(body, options = {}) {
   if (ids.length) {
     query._id = { $in: ids };
   }
-  if (keyword && !options.keywordFields) {
+  if (keyword && !keywordFields.length) {
     throw new Error('No keyword search has been configured for this API call');
   }
-  if (keyword && options.keywordFields) {
-    query.$or = options.keywordFields.map((field) => {
-      return {
-        [field]: keyword,
-      };
-    });
+  if (keyword) {
+    query.$or = [];
+    const reg = RegExp(escapeRegExp(keyword), 'i');
```

Comment:
Perhaps the order is wrong here ? 
Confirm first its not an objectId to avoid search the db with unnecessary

Reply:
I think I get what you're saying now... however looking at `ObjectId.isValid` it's just checking for an ascii string of length 12 or 24. If the keyword happened to be exactly that length we still want to check on the keyword fields for it I think, otherwise it won't match